cfpb/cfgov-refresh
# DF-1607: Processor update and tests

This PR addresses DF-1607 by fixing the content processors to match the WordPress API output. It also adds the basis for unit tests for the content processors along with some basic tests for the changes made in this PR. The unit tests are very, very, very far from complete, and only test the changes this PR makes, but they're a start.

**Additions**
 
- Test runner and unit tests for content processors
 
**Changes**
 
- Changed `wordpress_orgmember_processor.py`, `wordpress_post_processor.py`, and `wordpress_view_processor.py` to match WordPress API output for custom fields.
 
**Testing**

From the `_tests/processor_tests` directory,

1. Run `pip install -r requirements.txt` (installs Mock if it is not already installed the current virtualenv).
2. Run `python test_processors.py`

It would also probably be a good idea to run a Sheer reindex with these processors. From the cfgov-refresh root:

- `sheer index --reindex` 

And then browse to relevant pages on the site. These are:

- http://localhost:7000/the-bureau/bureau-structure/ (Look for single and multiple titles under names)
- http://localhost:7000/blog/ (The hero should display as normal)
- http://localhost:7000/newsroom/ (Look for links in the "Featured Topic")

**Review**
 
- @dpford 
- @kurtw
- @jimmynotjim 
- @anselmbradford 
- @sebworks 

**Notes**

Prior to this change, the `cfpb_hero` type had only one `related_post` being indexed in Elasticsearch by the content processor, meaning any additional `related_post` values were being lost in indexing. This PR changes that, so that all possible `related_post` values are being indexed. 

To preserve existing behavior, `_includes/hero.html` is now selecting the first `related_post` from the `related_posts` list. However, given that heros can potentially have multiple related posts, it might be worth reconsidering the way the hero macros handle related posts.
